client-v2/src/components/FrontsEdit/CollectionComponents/Collection.tsx

```diff
@@ -192,15 +194,20 @@ class Collection extends React.Component<CollectionProps> {
                 <Button
                   size="l"
                   priority="primary"
-                  onClick={() => publish(id, frontId)}
+                  onClick={() => {
```

Comment:
The logic looks good here! But there are two things I should mention:

- this will cause a warning (and a memory leak, although it will be short lived!) if the component is unmounted before the operation completes -- see https://www.robinwieruch.de/react-warning-cant-call-setstate-on-an-unmounted-component for an example solution.
- Adding this as a class method rather than inline with the JSX might make the component easier to read.

Reply:
👍